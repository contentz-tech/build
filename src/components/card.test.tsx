/** * @jest-environment jsdom */
import '@testing-library/jest-dom/extend-expect'
import { jsx } from "@emotion/core";
import { render } from "@testing-library/react";
import Card from "./card";

describe("Card", () => {
  test("Non Selectable", () => {
    const { asFragment, getByText } = render(
      <Card>This is the content of the card.</Card>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/This is the content of the card/)).toBeInTheDocument();
    expect(getByText(/This is the content of the card/))
      .toHaveStyle("user-select: none");
  });

  test("Selectable", () => {
    const { asFragment, getByText } = render(
      <Card selectable>This is the content of the card.</Card>
    );

    expect(asFragment()).toMatchSnapshot();
    expect(getByText(/This is the content of the card/)).toBeInTheDocument();
    expect(getByText(/This is the content of the card/))
      .toHaveStyle("user-select: text");
  });
});
