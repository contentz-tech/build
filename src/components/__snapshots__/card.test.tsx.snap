// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Non Selectable 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: rgba(0,0,0,0.12) 0px 2px 5px 0px;
  background-color: white;
  border-radius: 5px;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin: 1.5em -2em 3em;
  padding: 1em 2em;
}

@media (max-width:720px) {
  .emotion-0 {
    box-shadow: none;
    border-radius: 0;
    margin: 0;
    padding: 1em 0.5em;
  }
}

<div
    class="emotion-0"
  >
    This is the content of the card.
  </div>
</DocumentFragment>
`;

exports[`Card Non Selectable 2`] = `
<DocumentFragment>
  .emotion-0 {
  box-shadow: rgba(0,0,0,0.12) 0px 2px 5px 0px;
  background-color: white;
  border-radius: 5px;
  -webkit-transition: all 0.2s ease 0s;
  transition: all 0.2s ease 0s;
  -webkit-user-select: text;
  -moz-user-select: text;
  -ms-user-select: text;
  user-select: text;
  margin: 1.5em -2em 3em;
  padding: 1em 2em;
}

@media (max-width:720px) {
  .emotion-0 {
    box-shadow: none;
    border-radius: 0;
    margin: 0;
    padding: 1em 0.5em;
  }
}

<div
    class="emotion-0"
  >
    This is the content of the card.
  </div>
</DocumentFragment>
`;
